app/routes/v2/standards/index.js

```diff
@@ -44,3 +44,22 @@ module.exports = [{
     }
   }
 }]
+
+const getAllStandards = async (request, error) => {
+  const applyJourney = await cache.get('apply-journey', request.yar.id)
+  let standards = null
+  if (error && applyJourney.standards) {
+    standards = applyJourney.standards
+  } else {
+    await sendStandardsRequestMessage({ sbi: applyJourney.selectedSbi.sbi, organisationId: applyJourney.selectedSbi.organisationId, callerId: applyJourney.callerId }, request.yar.id)
+    const response = await getPollingResponse(request.yar.id, '/standards')
+    if (response) {
+      console.info('Standards request received', response)
+      await cache.update('apply-journey', request.yar.id, { agreementNumber: response.agreementNumber ?? generateAgreementNumber(), standards: response.standards })
```

Comment:
Maybe do the check outside of the update cache.

`const agreementNumber = response.agreementNumber ?? generateAgreementNumber()`